src/Type/ValueOfType.php

```diff
@@ -50,6 +51,13 @@ public function isResolvable(): bool
 	protected function getResult(): Type
 	{
 		if ($this->type->isEnum()->yes()) {
+			if (
+				$this->type instanceof TemplateType
+				&& $this->type->getBound()->equals(new ObjectType('BackedEnum'))
```

Comment:
should this be the other way arround?
```suggestion
				&& new ObjectType('BackedEnum')->isSuperTypeOf($this->type->getBound())->yes()?
```

Reply:
just had a look into the code-base. it seems we have spots where a `Bound` is compared using `equals` and sometimes it uses `isSuperTypeOf`.

I am not sure whether thats on purpose or an oversight

Reply:
isSuperTypeOf is better here indeed, in case someone use 
```
interface SuperBackedEnum extends \BackedEnum
```
and a template of SuperBackedEnum.

test added